src/firefly/js/visualize/ChartsTableViewPanel.jsx

```diff
@@ -402,14 +413,24 @@ class ChartsPanel extends React.Component {
     }
 
     renderToolbar() {
+        const {expandable, expandedMode} = this.props;
```

Comment:
IMO and without full knowledge of this component, I think breaking the many render functions into private functional components will make the main/public component looks cleaner.  Also from a performance perspective, it may not have to re-render.
i.e.  
function Toolbar({expandable, expandedMode}) {...}
then use <Toolbar> in place of renderToolbar